velox/exec/NestedLoopJoinProbe.cpp

```diff
@@ -684,4 +700,46 @@ RowVectorPtr NestedLoopJoinProbe::getBuildMismatchedOutput(
       pool(), outputType_, nullptr, numUnmatched, std::move(projectedChildren));
 }
 
+RowVectorPtr NestedLoopJoinProbe::getOutputLeftSemiJoinImpl() {
+  VELOX_CHECK_NOT_NULL(input_);
+
+  bool matched = false;
+  numOutputRows_ = 0;
+
+  if (isBuildSideEmpty()) {
+    matched = false;
+  } else if (joinCondition_ == nullptr) {
+    matched = true;
+  } else {
+    for (auto i = buildRow_; i < decodedFilterResult_.size(); ++i) {
+      if (isJoinConditionMatch(i)) {
```

Comment:
I think we could have a fast check on this if any row has matched instead of using a loop here,